google-cloud-bigquery/lib/google/cloud/bigquery/policy.rb

```diff
@@ -0,0 +1,341 @@
+        #     account.
+        #   * `allAuthenticatedUsers`: A special identifier that represents anyone who is authenticated with a Google
+        #     account or a service account.
+        #   * `user:<emailid>`: An email address that represents a specific Google account. For example,
+        #     `alice@example.com`.
+        #   * `serviceAccount:<emailid>`: An email address that represents a service account. For example,
+        #     `my-other-app@appspot.gserviceaccount.com`.
+        #   * `group:<emailid>`: An email address that represents a Google group. For example, `admins@example.com`.
+        #   * `deleted:user:<emailid>?uid=<uniqueid>`: An email address (plus unique identifier) representing a user
+        #     that has been recently deleted. For example, `alice@example.com?uid=123456789012345678901`. If the user
+        #     is recovered, this value reverts to `user:<emailid>` and the recovered user retains the role in the
+        #     binding.
+        #   * `deleted: serviceAccount:<emailid>?uid=<uniqueid>`: An email address (plus unique identifier) representing
+        #     a service account that has been recently deleted. For example,
+        #     `my-other-app@appspot.gserviceaccount.com?uid=123456789012345678901`. If the service account is undeleted,
+        #     this value reverts to `serviceAccount:<emailid>` and the undeleted service account retains the role in
+        #     the binding.
+        #   * `deleted:group:<emailid>?uid=<uniqueid>`: An email address (plus unique identifier) representing a Google
+        #     group that has been recently deleted. For example, `admins@example.com?uid=123456789012345678901`. If the
+        #     group is recovered, this value reverts to `group:<emailid>` and the recovered group retains the role in
+        #     the binding.
+        #   * `domain:<domain>`: The G Suite domain (primary) that represents all the users of that domain. For example,
+        #     `google.com` or `example.com`.
+        #
+        # @return [Binding, nil] The binding object that was added to the policy. This object is mutable and may be used
+        #   to add or remove members within the context of a block passed to {Table.update_policy}.
+        #
+        # @example Update a mutable policy with {Table#update_policy}.
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #   dataset = bigquery.dataset "my_dataset"
+        #   table = dataset.table "my_table"
+        #
+        #   table.update_policy do |p|
+        #     p.grant members: "user:viewer@example.com", role: "roles/viewer"
+        #   end # 2 API calls
+        #
+        def grant members:, role:
+          new_binding = Binding.new role, members
+          if (i = @bindings.find_index { |b| b.role == role })
+            @bindings[i] = new_binding
+          else
+            @bindings << new_binding
+          end
```

Comment:
We shouldn't replace a binding if it exists. So we want something like:

```ruby
b = @bindings.find { |b| b.role == role }
if b
  b.members.concat(Array(members))
else
  b = Binding.new role, members
  @bindings << b
end
b
```

Reply:
Done.